pkg/apiserver/registry/networkpolicy/clustergroupmember/rest.go

```diff
@@ -70,9 +84,50 @@ func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions)
 		memberList.EffectiveMembers = effectiveMembers
 	}
 	memberList.Name = name
-	return memberList, nil
+	memberList.TotalMembers = int64(len(memberList.EffectiveMembers))
+	err = paginateMemberList(memberList, pageInfo)
+	return memberList, err
+}
+
+func (r *REST) NewGetOptions() (runtime.Object, bool, string) {
+	return &controlplane.PaginationGetOptions{}, false, ""
 }
 
 func (r *REST) NamespaceScoped() bool {
 	return false
 }
+
+// paginateMemberList returns paginated results if rational options are provided.
+// Always return nil error, absorbs logical error provided in query by returning full list of members.
+func paginateMemberList(memberList *controlplane.ClusterGroupMembers, pageInfo paginationInfo) error {
+	if memberList.TotalMembers == 0 {
+		memberList.TotalPages = 0
+		memberList.CurrentPage = 0
+		return nil
+	}
+	if pageInfo.pageLimit > 0 && pageInfo.pageNumber > 0 {
+		pageInfo.totalPages = int(math.Ceil(math.Max(float64(len(memberList.EffectiveMembers))/float64(pageInfo.pageLimit), float64(pageInfo.totalPages))))
```

Comment:
Isn't it the case that `pageInfo.totalPages` is always equal to 1 when this function is called?
I am actually not entirely sure why `totalPages` is part of `paginationInfo`. It seems to be more of an output of this function and less of an input?
I was wondering if this would work just as well:
```go
pageInfo.totalPages = (len(memberList.EffectiveMembers) + pageInfo.pageLimit - 1) / pageInfo.pageLimit
```
I get a bit uncomfortable when we start casting things to floats :)

Reply:
That is very inspiring! I also modified the logic for retrieving targeted member list trunk, b/c it also involved float conversions.